### PR TITLE
imxrt: use UART_CONSOLE to setup early console

### PR DESCRIPTION
**Pull request by gerard5**
JIRA: RTOS-118

## Description

On `imxrt117x` the `UART_CONSOLE_PLO`, and `UART_CONSOLE` environment variables do not configure the (so-called) early plo console (default port is hardcoded) side effect of this may observed when plo starts:
before

Hardcoded values are assigned to `UART1` which is not the default console port for the current project being developed on `imxrt117x`, for ease of use with `BOARD_CONFIG`,  `UART_CONSOLE` or `UART_CONSOLE_PLO` may be used to assign a default port per project. I assume the macros provided are per each `console.c`, but may be moved to `peripherals.h`.

This commit fixes that, so with default `UART_CONSOLE` from `peripherals.h` which is `UART11` it looks like:
after

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (imxrt1176-custom, imxrt1064-evalkit).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by agkaminski** [comment]
Please rebase on the current master

**Review by agkaminski — APPROVED** [review]
We prefer runtime selections over macros because we try to reduce macros usage to the minimum, but this is too neat for me to not approve it.

**Comment by gerard5** [comment]
@agkaminski there will be one change to `CONSOLE_BAUD`, (baudrate is hadrcoded to 115200), but it is desirable to support `9600`, `19200` (opto), and `115200` so let me add this before merging.

File: hal/armv7m/imxrt/10xx/console.c
Comment (agkaminski) [reply hal/armv7m/imxrt/10xx/console.c:42]: Can't we just pass an integer? Redefining speed_t hurts me somewhat

File: hal/armv7m/imxrt/10xx/console.c
Comment (agkaminski) [inline hal/armv7m/imxrt/10xx/console.c:94]: Maybe define another constant on top of the file and use it here? #if..#else...#endif inside of switch condition looks really bad

File: hal/armv7m/imxrt/10xx/console.c
Comment (agkaminski) [inline hal/armv7m/imxrt/10xx/console.c:86]: Stale comment above

File: hal/armv7m/imxrt/10xx/console.c
Comment (gerard5) [reply hal/armv7m/imxrt/10xx/console.c]: Done.

File: hal/armv7m/imxrt/10xx/console.c
Comment (gerard5) [reply hal/armv7m/imxrt/10xx/console.c:92]: Ack.

Comment (agkaminski) [reply hal/armv7m/imxrt/10xx/console.c:92]: I meant "/* Set 115200 baudrate */" in line 91

File: hal/armv7m/imxrt/10xx/console.c
Comment (agkaminski) [reply hal/armv7m/imxrt/10xx/console.c:48]: Sorry, my comment is about the same line, you might've missed it

File: hal/armv7m/imxrt/10xx/console.c
Comment (gerard5) [reply hal/armv7m/imxrt/10xx/console.c:92]: Right, updated the comment. Missed that one. Thanks.

**Review by agkaminski — APPROVED** [review]

**Comment by gerard5** [comment]
Updated to the changes introduced by PR: https://github.com/phoenix-rtos/phoenix-rtos-devices/pull/150 and adding @MaciejPurski to the reviewers. Please verify if the PR works for your specific cases on `imxrt1064`.

**Comment by gerard5** [comment]
Following the changes in `plo`, I'll introduce early console setup in the kernel based on this PR.

File: hal/armv7m/imxrt/10xx/console.c
Comment (agkaminski) [inline hal/armv7m/imxrt/10xx/console.c:95]: What is set in default case? Comment would be helpful

File: hal/armv7m/imxrt/10xx/console.c
Comment (agkaminski) [inline hal/armv7m/imxrt/10xx/console.c:29]: I like previous version better. `undef` is one bridge too far

File: hal/armv7m/imxrt/10xx/console.c
Comment (gerard5) [reply hal/armv7m/imxrt/10xx/console.c:97]: Added comment about default fallback value.

File: hal/armv7m/imxrt/10xx/console.c
Comment (gerard5) [reply hal/armv7m/imxrt/10xx/console.c]: Ok.

**Review by agkaminski — APPROVED** [review]
LGTM

File: hal/armv7m/imxrt/10xx/console.c
Comment (MaciejPurski) [inline hal/armv7m/imxrt/10xx/console.c:29]: `CONSOLE_BAUDRATE` is already in use in phoenix-rtos-kernel:
https://github.com/phoenix-rtos/phoenix-rtos-kernel/blob/071fe90ba4373afedad6296db12d2417669299bd/hal/armv7m/imxrt/10xx/console.c#L21
I use it in `BOARD_CONFIG` in order to set the kernel early console baudrate, so your code triggers a warning in this case. However, if you add the same mechanism to the kernel this `CONSOLE_BAUDRATE` in `BOARD_CONFIG` won't be needed, so there won't be a collision. Therefore, it's fine for me.

**Review by MaciejPurski — APPROVED** [review]
Works as expected. The issue with possible `CONSOLE_BAUDRATE` redefinition won't be problematic once you implement the same mechanism in kernel, so it's not a merge-blocker for me.
Approved.

**Review by HBuczynski — APPROVED** [review]

File: hal/armv7m/imxrt/10xx/console.c
Comment (gerard5) [reply hal/armv7m/imxrt/10xx/console.c:29]: Yes, it will work like this: any of the `n` uarts can define a custom `UARTn_BAUDRATE` in `BOARD_CONFIG` instead of a single `CONSOLE_BAUDRATE`, this is now consistent with `imxrt-multi` and `plo` (current PR), so the `kernel`s early console is underway  (I'm testing it now). That's right, I changed the logic in the kernel in a similar way to this PR.

**Pull request merged**
